test/realm_object_test.dart

```diff
@@ -414,32 +414,23 @@ Future<void> main([List<String>? args]) async {
 
   final epochZero = DateTime.fromMillisecondsSinceEpoch(0, isUtc: true);
 
-  bool _canCoreRepresentDateInJson(DateTime date) {
-    // Core has a bug where negative and zero dates are not serialized correctly to json.
-    // https://jira.mongodb.org/browse/RCORE-1083
-    if (date.compareTo(epochZero) <= 0) {
-      return Platform.isMacOS || Platform.isIOS;
-    }
-
-    // Very large dates are also buggy on Android and Windows
-    if (date.compareTo(DateTime.utc(10000)) > 0) {
-      return Platform.isMacOS || Platform.isIOS || Platform.isLinux;
-    }
-
-    return true;
-  }
-
   void expectDateInJson(DateTime? date, String json, String propertyName) {
     if (date == null) {
       expect(json, contains('"$propertyName":null'));
-    } else if (_canCoreRepresentDateInJson(date)) {
+    } else {
       expect(json, contains('"$propertyName":"${date.toCoreTimestampString()}"'));
     }
   }
 
   final dates = [
-    DateTime.utc(1970).add(Duration(days: 100000000)),
-    DateTime.utc(1970).subtract(Duration(days: 99999999)),
+    // BUG: core timestamp serialization does not support years
+    // very far from 1970 currently currently
```

Comment:
```suggestion
    // very far from 1970 currently
```

Reply:
Is this a bug or by purpose?

Reply:
I believe it is a bug. Realm represent timestamps as 64 bit seconds offset from epoch (1970) and 32 bit nano-seconds. Hence realm can represent offsets from epoch up to 9223372036854689408 days, yet realm-core cannot serialise that.

Reply:
I elaborated on the comment to make it more precise. The reason the PR is still in draft was mostly to raise this issue with core first.

Reply:
I have raised https://github.com/realm/realm-core/issues/6892